x/margin/types/pool.go

```diff
@@ -2,22 +2,117 @@ package types
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ammtypes "github.com/elys-network/elys/x/amm/types"
 )
 
 func NewPool(poolId uint64) Pool {
 	return Pool{
-		AmmPoolId:                    poolId,
-		Health:                       sdk.ZeroDec(),
-		Enabled:                      true,
-		Closed:                       false,
-		ExternalLiabilities:          sdk.ZeroInt(),
-		ExternalCustody:              sdk.ZeroInt(),
-		NativeLiabilities:            sdk.ZeroInt(),
-		NativeCustody:                sdk.ZeroInt(),
-		InterestRate:                 sdk.ZeroDec(),
-		NativeAssetBalance:           sdk.ZeroInt(),
-		ExternalAssetBalance:         sdk.ZeroInt(),
-		UnsettledExternalLiabilities: sdk.ZeroInt(),
-		BlockInterestNative:          sdk.ZeroInt(),
+		AmmPoolId:    poolId,
+		Health:       sdk.ZeroDec(),
+		Enabled:      true,
+		Closed:       false,
+		InterestRate: sdk.ZeroDec(),
+		PoolAssets:   []PoolAsset{},
+	}
+}
+
+// Update the asset balance
+func (p *Pool) UpdateBalance(ctx sdk.Context, assetDenom string, amount sdk.Int, isIncrease bool) error {
+	for i, asset := range p.PoolAssets {
+		if asset.AssetDenom == assetDenom {
+			if isIncrease {
+				p.PoolAssets[i].AssetBalance = asset.AssetBalance.Add(amount)
+			} else {
+				p.PoolAssets[i].AssetBalance = asset.AssetBalance.Sub(amount)
```

Comment:
Shouldn't we add negative amount checker?

Reply:
@jelysn this can be negative as it is the balance of margin pool. Real pool balance will be amm pool + margin pool.

Reply:
Okay